tests/schemas/rules/boolean/value/date.json

```diff
@@ -0,0 +1,177 @@
+[
+  {
+    "description": "A date answer equal to a fixed date",
+    "rule": {
+      "==": [
+        {
+          "date": [
+            {
+              "source": "answers",
+              "identifier": "answer-1"
+            }
+          ]
+        },
+        {
+          "date": ["2021-01-01"]
+        }
+      ]
+    }
+  },
+  {
+    "description": "A date answer equal to a dynamic date",
+    "rule": {
+      "==": [
+        {
+          "date": [
+            {
+              "source": "answers",
+              "identifier": "answer-1"
+            }
+          ]
+        },
+        {
+          "date": ["now"]
```

Comment:
Can we have anything other than now as a dynamic date?

Reply:
No, `now` or an actual data via a source or `self` which would be resolved to a string date being calling the date operator.